rfc-120-csv-preview.md

```diff
@@ -0,0 +1,134 @@
+
+- Using a top-level prefix for CSV Preview is at-odds with the behaviour of other top-level prefixes (`/info`, `/api/content`), since it only works for specific URLs, and doesn't apply to GOV.UK content in general.
+
+### Improve the feature in Whitehall (frontend)
+
+This would involve rewriting the existing CSV Preview feature so that it can work for documents published elsewhere, instead of being coupled to the attachment model in Whitehall.
+
+> We could make the feature work for attachments published by others apps, using the proposed URL structure, which gives us the parent document and the `asset-id` of attachment being previewed.
+
+- This would mean continued reliance of the Whitehall (frontend) codebase and the `/government` prefix route, which has been identified as [tech debt][government-prefix].
+
+- Not using specific routes for each attachment preview means we won't be able to do think like automatic redirects (if the base path of the parent document changes).
+
+### Using sub-routes of the parent document for the routing
+
+The Publishing API supports any number of `routes` for a content item. Normally, this will just be a single route, equal to the `base_path` of the content item. We could add an extra route or routes to support CSV Preview.
+
+> Any additional routes that exist (and which are not redirects) will use the same rendering app as the 'main' route. This fits with the proposal to re-implement CSV Preview in Government Frontend.
+
+> We could use a prefix sub-route (`.../*`), or specific sub-routes for each attachment. Using a prefix sub-route is less effort, but the behaviour is ambiguous if an attachment is not found or associated with the document.
+
+- This would have a lower implementation effort for individual publishing apps, as opposed to creating a separate content item for each attachment preview, which could get out-of-sync with the parent document.
+
+- Most of the lifecycle management, such as unpublishing and redirects, would be done by the Publishing API. However, we would need to address a number of issues with sub-routes in the Publishing API.
+
+  - Publishing API does not clean up sub-routes when they are no longer present in the content item. It _will_ handle redirects for them if the parent base path changes, but that's it.
+
+  - Publishing API does not propagate changes in document state to its sub-routes. For example, a remove/redirect on the parent document does nothing to its sub-routes.
+
+- It's surprising to see routes that aren't related to the actual content item being defined as part of the content item itself. This would be a new pattern for GOV.UK, which someone working on the feature needs to learn.
```

Comment:
```suggestion
- It's surprising to see routes that aren't related to the actual content item being defined as part of the content item itself. This would be a new pattern for GOV.UK, which someone working on the feature would need to learn.
```

Reply:
Fixed 👍